lib/block_view.go

```diff
@@ -4053,7 +4054,28 @@ func (bav *UtxoView) ConnectBlock(
 		// which a miner is trying to spam the network, which should generally never happen.
 		utxoOpsForTxn, totalInput, totalOutput, currentFees, err := bav.ConnectTransaction(txn, txHash, 0, uint32(blockHeader.Height), int64(blockHeader.TstampNanoSecs), verifySignatures, false)
 		_, _ = totalInput, totalOutput // A bit surprising we don't use these
-		if err != nil {
+		// After the PoS cutover, we need to check if the transaction is a failing transaction.
+		if blockHeight >= uint64(bav.Params.ForkHeights.ProofOfStake2ConsensusCutoverBlockHeight) {
+			if desoBlock.TxnConnectStatusByIndex == nil {
+				return nil, fmt.Errorf("ConnectBlock: TxnConnectStatusByIndex is nil at block height %d", blockHeight)
+			}
+			txnConnects := (txIndex == 0 && txn.TxnMeta.GetTxnType() == TxnTypeBlockReward) ||
+				desoBlock.TxnConnectStatusByIndex.Get(txIndex-1)
+			if !txnConnects {
+				if err == nil {
+					return nil, errors.Wrapf(err, "ConnectBlock: txn #%d should not connect based on "+
+						"TxnConnectStatusByIndex but err is nil", txIndex)
+				}
+				var burnFee, utilityFee uint64
+				utxoOpsForTxn, burnFee, utilityFee, err = bav._connectFailingTransaction(txn, uint32(blockHeader.Height), verifySignatures)
+				if err != nil {
+					return nil, errors.Wrapf(err, "ConnectBlock: error connecting failing txn #%d", txIndex)
+				}
+				_, _ = burnFee, utilityFee // TODO: figure out what we're supposed to do with these ones.
+			} else if err != nil {
+				return nil, errors.Wrapf(err, "ConnectBlock: error connecting txn #%d", txIndex)
+			}
+		} else if err != nil {
 			return nil, errors.Wrapf(err, "ConnectBlock: error connecting txn #%d", txIndex)
 		}
```

Comment:
Is there a way to simplify the if block nesting here? I just took a stab at it:
```suggestion
		if blockHeight < uint64(bav.Params.ForkHeights.ProofOfStake2ConsensusCutoverBlockHeight) {
			// If the block is a PoW block, and we fail to connect the txn, then 
			if err != nil {
				return nil, errors.Wrapf(err, "ConnectBlock: error connecting txn #%d", txIndex)
			}
			// Happy path for PoW block
			continue
		}

		// At this point, we know the block is a PoS block. We need to check if the transaction
		// is a failing transaction.
		if desoBlock.TxnConnectStatusByIndex == nil {
			return nil, fmt.Errorf("ConnectBlock: TxnConnectStatusByIndex is nil at block height %d", blockHeight)
		}
		
		// Compute the PoS txn's connect status
		txnConnects := (txIndex == 0 && txn.TxnMeta.GetTxnType() == TxnTypeBlockReward) ||
			desoBlock.TxnConnectStatusByIndex.Get(txIndex-1)
			
		// Handle the case where the PoS txn is supposed to connect
		if txnConnects {
			// If the txn is intended to connect but fails, then return an error
			if err != nil {
				return nil, errors.Wrapf(err, "ConnectBlock: error connecting txn #%d", txIndex)
			}
			// Happy path whether the txn connects
		}

		// At this point, the PoS txn is not supposed to connect. The ConnectTransaction call above should
		// have errored
		if err == nil {
			return nil, errors.Wrapf(err, "ConnectBlock: txn #%d should not connect based on "+
				"TxnConnectStatusByIndex but err is nil", txIndex)
		}
		
		var burnFee, utilityFee uint64
		utxoOpsForTxn, burnFee, utilityFee, err = bav._connectFailingTransaction(txn, uint32(blockHeader.Height), verifySignatures)
		if err != nil {
			return nil, errors.Wrapf(err, "ConnectBlock: error connecting failing txn #%d", txIndex)
		}
		
		_, _ = burnFee, utilityFee // TODO: figure out what we're supposed to do with these ones.
```

Reply:
can't use a continue here unfortunately 😢 - will look at other options to simplify it.